### PR TITLE
[SRBT-249] Set up Telemetry

### PR DESCRIPTION
**Pull request by ChristopherPark3**
# [SRBT-249] Set up Telemetry
**Changes**

- Sets up PostHog session replays for the first minute (can easily change)
- Implements the beginning of a session recording in `all-set.tsx`, the moment the user presses the button to continue to their profile page.
- The recording is then stopped based on an effect in `/[handle]/page.tsx` after a `60 sec` timeout has completed.
  - Noting that when the `/[handle]/page.tsx` component unmounts (i.e. in the event of page navigation), the timeout may be ended as a result of the clean up function. Thus, recordings can be cut short. 
  - This is how it plays out in my head theoretically, but upon testing, things seem to be working fine.
- Installs `posthog-js` and `posthog-node` packages

** Newest Changes ** 
- Moved the effect for managing the session replay timer to `posthog-provider.tsx`
- Added a feature flag for the session replays, disabling them on dev
  - I just don't want to keep doing recordings everytime i make a new account locally.
  - * There is a plan to create multiple environments on PostHog (prod, dev, staging), just like our other services.

### CODE REVIEW

File: src/components/onboarding/signup/all-set.tsx [hunk 1]
```diff
@@ -23,6 +24,11 @@ const AllSet = () => {
     return () => clearTimeout(timer);
   }, []);
 
+  const handleContinue = () => {
+    router.push(`/${userData.handle}`)
```
Comment (Cutaiar) [inline src/components/onboarding/signup/all-set.tsx:28]: Are you sure you don't want to start the recording before router push? I wonder if this creates a race condition where the push finishes before the recording is started?

File: src/app/posthog-provider.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  }, []);
+
+  return (
+    <PostHogProvider client={posthog}>
+      <PostHogIdentityWrapper>{children}</PostHogIdentityWrapper>
+    </PostHogProvider>
+  );
+}
+
+const PostHogIdentityWrapper = ({ children }: { children: ReactNode }) => {
```
Comment (Cutaiar) [inline src/app/posthog-provider.tsx:28]: little tsdoc about what this does for our future selves?

File: src/app/posthog-provider.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import { PostHogProvider } from 'posthog-js/react';
+import { ReactNode, useEffect } from 'react';
+
+import { useAuth } from '@/hooks';
+import { env } from '@/lib/env';
+
+export function PHProvider({ children }: { children: ReactNode }) {
+  useEffect(() => {
+    posthog.init(env.NEXT_PUBLIC_POSTHOG_KEY, {
+      api_host: process.env.NEXT_PUBLIC_POSTHOG_HOST,
```
Comment (Cutaiar) [inline src/app/posthog-provider.tsx:13]: probably want to access this from `@/lib/env` no?

File: src/app/posthog-page-view.tsx
Comment (Cutaiar) [inline src/app/posthog-page-view.tsx:1]: Have you considered making this a hook like `usePostHogPageView` instead? This is essentially a component that encapsulates a few hook calls, useEffect, and returns null.

File: src/app/[handle]/page.tsx [hunk 4]
```diff
@@ -48,6 +49,17 @@ const ProfilePage = ({ params }: { params: { handle: string } }) => {
     },
   });
 
+  useEffect(() => {
```
Comment (Cutaiar) [inline src/app/[handle]/page.tsx:52]: Perhaps a little note above the effect describing what this does?

**Review by Cutaiar — APPROVED** [review]
Approved with suggestions 🦔

File: src/components/onboarding/signup/all-set.tsx (hunk 1 above)
Comment (ChristopherPark3) [reply src/components/onboarding/signup/all-set.tsx:28]: Ahh that's a good point

File: src/app/posthog-provider.tsx (hunk 3 above)
Comment (ChristopherPark3) [reply src/app/posthog-provider.tsx:13]: nice catch thanks!

File: src/app/posthog-page-view.tsx
Comment (ChristopherPark3) [reply src/app/posthog-page-view.tsx:1]: yea that's a good point, but i was just following what the PostHog documentation said to do

Comment (ChristopherPark3) [reply src/app/posthog-page-view.tsx:1]: one reason i see is to keep the layout a server component. if we make it a hook, it'll have to be a client component, no?

File: src/lib/flags.ts [hunk 5]
```diff
@@ -4,4 +4,5 @@ type FeatureFlags = {
 
 export const featureFlags: FeatureFlags = {
   gigs: false,
+  sessionReplay: process.env.NODE_ENV === 'development' ? false : true,
```
Comment (Cutaiar) [inline src/lib/flags.ts:7]:
```suggestion
  sessionReplay: process.env.NODE_ENV === 'production',
```

Why not just enable it in production? And not staging?

**Pull request merged**